client/implementation-vertx/src/main/java/io/smallrye/graphql/client/vertx/dynamic/VertxDynamicGraphQLClient.java

```diff
@@ -304,10 +330,10 @@ public void close() {
 
     private Uni<Response> executeSingleResultOperationOverHttp(JsonObject json) {
         return Uni.createFrom().completionStage(
-                webClient.postAbs(url)
+                url.get().subscribeAsCompletionStage().thenCompose(instanceUrl -> webClient.postAbs(instanceUrl)
```

Comment:
Could you stay with Uni all the way?

Reply:
No, vert.x webclient returns a CompletionStage, so I need to combine Unis and CompletionStages, this seems to me to be the least ugly way